server/src/main/java/com/thoughtworks/go/server/domain/user/FilterValidator.java

```diff
@@ -49,6 +54,18 @@ static void validateFilter(Map<String, DashboardFilter> current, DashboardFilter
         validateNamePresent(name);
         validateNameFormat(name);
         validateNameIsUnique(current, name);
+        validateState(filter.state());
+    }
+
+    private static void validateState(List<CaseInsensitiveString> state) {
+        if (state != null) {
+            if (state.size() >= 3)
```

Comment:
It might be better to use a `Set<String>` here instead, so there's no need to check `size()`, which feels like an anti-pattern; we shouldn't care about size, only about the validity of the values. Also, as the states are constants, no need to use `CaseInsensitiveString`.